framework/test/jest/unit/specs/modules/chain/dpos/round_delegates.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const BigNum = require('@liskhq/bignum');
+const { randomInt } = require('../../../../utils');
+const delegatePublicKeys = require('./delegate_publickeys.json');
+
+const delegateAccounts = delegatePublicKeys.map((pk, index) => {
+	const balance = new BigNum(randomInt(100, 1000));
+	const rewards = new BigNum(randomInt(100, 500));
+	return {
+		balance,
+		rewards,
+		fees: balance.sub(rewards),
+		publicKey: pk,
+		votedDelegatesPublicKeys: [`abc${index}`, `def${index}`, `xyz${index}`],
+	};
+});
+
+const missedDelegateCount = 5;
+
+// First Delegate forged 3 times
+const delegatesWhoForgedThrice = delegateAccounts.slice(0, 1);
+
+// 2nd-4th forges 2 times
+const delegatesWhoForgedTwice = delegateAccounts.slice(1, 4);
+
+// last 5 in the list
+const delegatesWhoMissed = delegateAccounts.slice(-missedDelegateCount);
+
+// rest of the delegates forged once
+const delegatesWhoForgedOnce = delegateAccounts.filter(
+	d =>
+		![
+			...delegatesWhoMissed.map(a => a.publicKey),
+			...delegatesWhoForgedTwice.map(a => a.publicKey),
+			...delegatesWhoForgedThrice.map(a => a.publicKey),
+		].includes(d.publicKey),
+);
+
+const delegatesWhoForged = [
+	...delegatesWhoForgedThrice,
+	...delegatesWhoForgedTwice,
+	...delegatesWhoForgedOnce,
+	...delegatesWhoForgedTwice,
+	...delegatesWhoForgedThrice,
+	...delegatesWhoForgedThrice,
+];
+
+const uniqueDelegatesWhoForged = delegatesWhoForged.filter(
+	(d, index) =>
+		delegatesWhoForged.findIndex(a => a.publicKey === d.publicKey) === index,
+);
+
+const delegateWhoForgedLast = delegatesWhoForged[delegatesWhoForged.length - 1];
+
+if (delegatesWhoForged.length !== delegateAccounts.length) {
```

Comment:
Having assertions on test data makes sense?

Reply:
Also @yatki's code and I think it's fine